synapse/storage/databases/main/user_directory.py

```diff
@@ -663,6 +663,42 @@ async def get_user_dir_rooms_user_is_in(self, user_id):
         users.update(rows)
         return list(users)
 
+    @cached()
+    async def get_shared_rooms_for_users(self, user_id, other_user_id):
+        """
+        Returns the rooms that a user is in.
+
+        Args:
+            user_id(str): Must be a local user
+
+        Returns:
+            list: user_id
+        """
+        SQL = """
+            SELECT p1.room_id
+            FROM users_in_public_rooms as p1
+            INNER JOIN users_in_public_rooms as p2
+                ON p1.room_id = p2.room_id
+                AND p1.user_id = ?
+                AND p2.user_id = ?
+            UNION
+            SELECT room_id
+            FROM users_who_share_private_rooms
+            WHERE
+                user_id = ?
+                AND other_user_id = ?;
+        """
+        rows = await self.db_pool.execute(
```

Comment:
We tend to use a txn function paired with `runInteraction` to execute DB queries, rather than running `execute` on the `db_pool` itself. See other examples in this file and elsewhere.